### PR TITLE
API changes preparation

### PR DESCRIPTION
**Pull request by Zlopez**
This is a basic structure for the API changes, there isn't any actual call to OpenNebula.

Signed-off-by: Michal Konečný <mkonecny@redhat.com>

### CODE REVIEW

File: duffy/api_v1/views.py [hunk 1]
```diff
@@ -59,32 +60,52 @@ def nodeget():
...
+                                Host.arch == get_arch,
+                                Host.flavor == get_flavor
+                                ).order_by(db.asc(Host.used_count)).limit(get_count).all()
 
-    if len(hosts) != get_count:
-        return 'Insufficient Nodes in READY State'
+        if len(hosts) != get_count:
+            return 'Insufficient Nodes in READY State'
+
+    elif type == Session.TYPE_OPEN_NEBULA:
```
Comment (Zlopez) [inline duffy/api_v1/views.py:74]: Here should be an ansible script execution and then adding the table row.

File: duffy/api_v1/views.py [hunk 2]
```diff
@@ -59,32 +60,52 @@ def nodeget():
...
-                host.save()
+            if sess.type == Session.TYPE_BARE_METAL:
+                # if any of the hosts fail we should return any in-progress hosts
+                # to the pool for reclamation
+                for host in sess.hosts:
+                    host.state = 'Active'
+                    host.save()
+            elif sess.type == Session.TYPE_OPEN_NEBULA:
+                # if any of the hosts fail, we should delete it
+                for host in hosts:
```
Comment (Zlopez) [inline duffy/api_v1/views.py:107]: Here should be call to delete the machine in OpenNebula.

File: duffy/api_v1/views.py [hunk 3]
```diff
@@ -108,11 +129,19 @@ def nodedone():
...
+    if session.type == Session.TYPE_BARE_METAL:
+        for host in session.hosts:
+            host.state = "Deprovision"
+            host.session = None
+            host.session_id = ''
+            host.save()
+    elif session.type == Session.TYPE_OPEN_NEBULA:
+        hosts = OpennebulaHost.query.filter(OpennebulaHost.session == session).all()
+        for host in hosts:
+            # Deprovision the hosts
```
Comment (Zlopez) [inline duffy/api_v1/views.py:141]: Here should be call to delete the machine in OpenNebula.

File: duffy/api_v1/views.py [hunk 4]
```diff
@@ -160,9 +196,15 @@ def inventory():
...
         rtn_sessions = []
         for session in sessions:
-            for host in session.hosts:
-                sch = HostSchema().dump(host)
-                rtn_sessions.append([sch.data['hostname'],sch.data['session']])
+            if session.type == Session.TYPE_BARE_METAL:
+                for host in session.hosts:
+                    sch = HostSchema().dump(host)
+                    rtn_sessions.append([sch.data['hostname'],sch.data['session']])
+            elif session.type == Session.TYPE_OPEN_NEBULA:
```
Comment (Zlopez) [inline duffy/api_v1/views.py:203]: Only return VM hosts if session key is provided.

**Comment by nphilipp** [comment]
Hi Michal! As you're probably aware of, we're rewriting Duffy from the ground up, therefore I'll close old pull requests.

**Pull request closed without merging**